mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -421,6 +435,18 @@ object LinearRegression extends DefaultParamsReadable[LinearRegression] {
    */
   @Since("2.1.0")
   val MAX_FEATURES_FOR_NORMAL_SOLVER: Int = WeightedLeastSquares.MAX_NUM_FEATURES
+
+  /** String name for "auto". */
+  private[regression] val AUTO = "auto"
```

Comment:
I modified here to follow the way in the companion object of `MultilayerPerceptronClassifier`. I don't know the discussion about whether to introduce constants ...